split seq2seq script into summarization & translation (#10611)
huggingface/transformers: examples/seq2seq/README.md

- `FSMTForConditionalGeneration` (translation only)
- `T5ForConditionalGeneration`

> `run_summarization.py` is a lightweight example of how to download and preprocess a dataset from the [🤗 Datasets](https://github.com/huggingface/datasets) library or use your own files (jsonlines or csv), then fine-tune one of the architectures above on it.

Reviewer: ```suggestion
`run_summarization.py`  and  `run_translation.py` are lightweight examples of how to download and preprocess a dataset from the [🤗 Datasets](https://github.com/huggingface/datasets) library or use your own files (jsonlines or csv), then fine-tune one of the architectures above on it.
```

> `run_sum.py`  and  `run_trans.py` are lightweight examples of how to download and preprocess a dataset from the [🤗 Datasets](https://github.com/huggingface/datasets) library or use your own files (jsonlines or csv), then fine-tune one of the architectures above on it.

Reviewer 2: ```suggestion
`run_sum.py` and `run_trans.py` are lightweight examples of how to download and preprocess a dataset from the [🤗 Datasets](https://github.com/huggingface/datasets) library or use your own files (jsonlines or csv), then fine-tune one of the architectures above on it.
```

And the same with MBart:

```bash
python examples/seq2seq/run_trans.py \
    --model_name_or_path facebook/mbart-large-en-ro  \
    --do_train \
    --do_eval \

>     --source_lang en_XX \
>     --target_lang ro_RO 

Reviewer 2: ```suggestion
```
these are already there, please scroll down a bit.

Reviewer 2: oh, I see, it did work, it just ordered my comments incorrectly.

    --dataset_name wmt16 \
    --dataset_config_name ro-en \

>     --source_lang en_XX \

Reviewer 2: here where they are already there.

Reviewer 2: hmm, weird I guess one can't delete lines in "suggestion" - I tried to delete those 2 lines.

Note, that depending on the used model additional language-specific command-line arguments are sometimes required. Specifically:

> * MBart models require different `{source,target}_lang` values, e.g. in place of `en` it expects `en_XX`, for `ro` it expects `ro_RO`. The full MBart specification for language codes can be looked up [here](https://huggingface.co/facebook/mbart-large-cc25)

Reviewer 2: ```suggestion
* MBart models require different `--{source,target}_lang` values, e.g. in place of `en` it expects `en_XX`, for `ro` it expects `ro_RO`. The full MBart specification for language codes can be looked up [here](https://huggingface.co/facebook/mbart-large-cc25)
```

* T5 requires:

   ```
       --source_lang en \

>        --target_lang ro \

Reviewer 3: T5 does not requires that, technically. It's required for the data preprocessing mostly.

Author: yup of course, but my current tradeoff (simplicity vs redundancy) makes them required (I raise a ValueError if these parameters aren't set), we can discuss the tradeoff but the doc is currently correct in this state.

Reviewer 3: Yes, and I like that they are needed arguments. But with this change they are not "additional language-specific command-line arguments are sometimes required" as said above. I would remove the paragraph about mBART before, or say it requires custom source/target language codes like `en_XX` instead of `en` and just mention the `source_prefix` for T5 here.
